### PR TITLE
SDKS-770 Properly cache the OKHttpClient

### PR DESCRIPTION
**Pull request by witrisna**

### CODE REVIEW

File: forgerock-auth/src/test/java/org/forgerock/android/auth/PersistentCookieTest.java [hunk 1]
```diff
@@ -27,9 +27,9 @@
 public class PersistentCookieTest extends BaseTest {
 
     @Before
-    public void setUpCookieStore() {
-        Config.getInstance().setCookieJar(SecureCookieJar.builder().context(context)
-                .build());
+    public void setupConfig() throws Exception {
```
Comment (rodrigoareis) [inline forgerock-auth/src/test/java/org/forgerock/android/auth/PersistentCookieTest.java:30]: Update year on the header

File: forgerock-auth/src/test/java/org/forgerock/android/auth/ServerConfigTest.java [hunk 2]
```diff
@@ -11,6 +11,7 @@
 
 import androidx.test.core.app.ApplicationProvider;
 
+import org.assertj.core.api.Assertions;
```
Comment (rodrigoareis) [inline forgerock-auth/src/test/java/org/forgerock/android/auth/ServerConfigTest.java:14]: Update year on the header

File: forgerock-core/src/main/java/org/forgerock/android/auth/NetworkConfig.java [hunk 3]
```diff
@@ -61,4 +64,12 @@ CookieJar getCookieJar() {
             return CookieJar.NO_COOKIES;
         }
     }
+
+    public String getIdentifier() {
```
Comment (rodrigoareis) [inline forgerock-core/src/main/java/org/forgerock/android/auth/NetworkConfig.java:68]: Missing documentation for public method

File: forgerock-core/src/main/java/org/forgerock/android/auth/OkHttpClientProvider.java [hunk 4]
```diff
@@ -39,7 +39,7 @@ public static OkHttpClientProvider getInstance() {
      * @return The OkHttpClient
      */
     OkHttpClient lookup(NetworkConfig networkConfig) {
-        OkHttpClient client = cache.get(networkConfig);
+        OkHttpClient client = cache.get(networkConfig.getIdentifier());
```
Comment (rodrigoareis) [inline forgerock-core/src/main/java/org/forgerock/android/auth/OkHttpClientProvider.java:42]: Update year on the header

**Review by rodrigoareis — COMMENTED** [review]
Overall changes looks good. Minor comments to be reviewed

**Review by spetrov — COMMENTED** [review]
Looks good to me! There are a couple of comments from Rodrigo you may want to address...

**Review by rodrigoareis — APPROVED** [review]
Changes looks good to me

**Review by spetrov — APPROVED** [review]

**Pull request merged**